### PR TITLE
Fixes/broken show when repository deleted

### LINKED ISSUE #19: The show page is broken when the repository is deleted
**Describe the bug**
A fatal error is raised because we try to access the repository to load the show page but it does not exists anymore.

**To Reproduce**
1. Go to 'Settings'
2. Click on 'Repository'
3. Delete the default repository
4. Create a new Git repository (in order to access the 'Pull Request' page)
5. Click on 'Pull Request'
6. Click on a pull request 

**Expected behavior**
A clean error should be displayed to the user.

### PR DESCRIPTION
**Pull request by xel1045**
# Description

This pull request add a warning if the user access the PR show and the repository is deleted.

Fixes #19.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

This PR add a new functional test to verify the bug and it was manually tested.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/models/pull.rb [hunk 1]
```diff
@@ -188,14 +188,19 @@ class Pull < ActiveRecord::Base
     state :cannot_be_merged
   end
 
+  # If you change this condition, please review the text `text_pull_broken`.
+  def broken?
+    ! repository
```
Comment (martinblanchette) [inline app/models/pull.rb]: In addition to this check, you should check if the physical repository exists (maybe by running Git commands on it).

File: app/views/pulls/_changes.html.erb [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
   <% diff = @pull.diff(@changes_from) %>
 
-  <% unless diff&.empty? %>
+  <% if diff && ! diff.empty? %>
```
Comment (martinblanchette) [inline app/views/pulls/_changes.html.erb]:
```suggestion
  <% unless diff&.empty? %>
```

File: app/views/pulls/show.html.erb [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 <h2><%= l(:label_pull_request) %> #<%= @pull.id %></h2>
 
+<% if @pull.broken? %>
+  <p class="warning"><span class="icon icon-lock"><%= l(:text_pull_broken) %></span></p>
```
Comment (martinblanchette) [inline app/views/pulls/show.html.erb]: I question your choice of the icon (lock) in relation to the type of message. You should maybe use a warning or broken link kind of icon OR use an "alert" type of message.

File: app/views/pulls/_changes.html.erb (hunk 2 above)
Comment (xel1045) [reply app/views/pulls/_changes.html.erb]: I've tested it, and it's not the same behaviour.

File: app/views/pulls/show.html.erb (hunk 3 above)
Comment (xel1045) [reply app/views/pulls/show.html.erb]: Done.

**Pull request merged**